### DISCUSSION TITLE
Is there a way to use Pytest with distest?

### DISCUSSION THREAD

**Discussion by NabeelUppel** [JakeCover/distest#61] [Q&A]
I'm trying to test my bot on travis ci by using your library. Is there anyway to use them together?

**Comment by Cobular** [comment]
Yea to travis, no to pytest (yet). Pytest plugin integration is the next thing on the to-do list, but it probably won't happen until summer because I have no time with school. However tho, take a look at the bash scripts in the root of the repo, I used to use those to test the bot on Travis before I moved it to GitHub actions. You can also look at the .github golfer for an example of the tests being run in actions.

**Reply by NabeelUppel** [reply]
Thank you, I'll give it a try. I'm sorry,I'm really new to all of this especially Travis CI, but will it be able to see if a test case has passed or not?

**Reply by Cobular** [reply]
Yep! Travis monitors if the thing you run returns 0 (success) or anything else (an error) and then will fail the run if anything non-zero was returned. Distest matches this system so it works just fine without distest.

**Reply by NabeelUppel** [reply]
Okay I got it up and running on Travis ci thanks to some of your examples and a lot of research. How would I go about setting up coverage and link it to coveralls,io for these tests?

**Reply by Cobular** [reply]
So I haven't setup automated code coverage things yet, but this should be doable. It seems like this https://github.com/TheKevJames/coveralls-python is the best way to send your code coverage results to coveralls, and you would first generate these with coverage.py. That library seems to just be a dropin replacement for the `python ` part of your commands, so in this project's `run_tests.sh` you'd replace the line `python example_target.py "$TARGET_TOKEN" &` with `coverage run example_target.py "$TARGET_TOKEN" &` to generate the coverage results. Then, you could use coveralls-python to upload them to coveralls.io as the next step in travis if the tests pass (and there should probably be a way to get a step to run whether or not the other steps run too, I just don't know enough travis to figure that out). 

It would basicaly be the equivalent of running
```bash
coverage run example_target.py "$TARGET_TOKEN"
coveralls
```
I think?

I'm not really sure if that helps too much, hopefully it does? I just don't have any direct experience with this, but hopefully this is enough to get started on that. I'll also let you know what I end up doing if I implement coverage on the library itself too.

**Reply by NabeelUppel** [reply]
Thanks dude for all your help and great library. I think its working now (although travis isn't updating coveralls but that's a separate issue). 
Quick questions, would I be able to create my own functions by using TestInterface functions?
Like using TestInterface's wait_for_message_in_channel. Basically define my own interface.function()?
